src/MudBlazor/Components/Snackbar/Snackbar.cs

```
@@ -30,30 +30,37 @@ internal Snackbar(SnackbarMessage message, SnackbarOptions options)
 
         internal void Init() => TransitionTo(SnackbarState.Showing);
 
-        internal async Task ClickedAsync(bool fromCloseIcon)
+        internal Task Clicked(bool fromCloseIcon)
         {
+            // You should only be able to interact with the snackbar once.
             if (State.UserHasInteracted)
-                return; // You should only be able to interact with the snackbar once.
+            {
+                return Task.CompletedTask;
+            }
 
             if (fromCloseIcon)
             {
                 // Invoke user-defined task when close button is clicked
-                if (State.Options.OnCloseButtonClick is not null)
-                {
-                    await State.Options.OnCloseButtonClick.Invoke(this);
-                }
+                //await State.Options.OnCloseButtonClick.Invoke(this);
+                State.Options.OnCloseButtonClick?.Invoke(this);
             }
             else
             {
                 // Do not start the hiding transition if no click action
-                if (State.Options.OnClick is null) return;
+                if (State.Options.OnClick is null)
+                {
+                    return Task.CompletedTask;
+                }
 
                 // Click action is executed only if it's not from the close icon
-                await State.Options.OnClick.Invoke(this);
+                //await State.Options.OnClick.Invoke(this);
+                State.Options.OnClick.Invoke(this);
```

Hmm, shouldn't `Clicked` still be async and this call be awaited? I might be  totally wrong

ActionClickedAsync and CloseIconClickedAsync could be non-async because they were directly calling another Task

---

Maybe someone could have a fresh pair of eyes look at it, but with the code as is, my async actions OnClick and OnCloseButtonClick are being awaited correctly in the case like this:
```
private void Show()
{
    Snackbar.Add("Close me using the close button.", Severity.Normal, config =>
    {
        config.RequireInteraction = true;
        config.OnCloseButtonClick = SayGoodbyeAsync;
        config.Action = "Adiós";
        config.OnClick = SayGoodbyeAsync;
    });
}

public async Task SayGoodbyeAsync(Snackbar snackbar)
{
    await Task.Delay(5000); // this is being awaited ok
    Snackbar.Add("Sorry to see you go!");
}
```
The snackbar will be dismissed immediately upon clicking the close button, with any attached task invoked and awaited in the background. I think that would work alright from a UX perspective. On the other hand, if in the `Clicked` method I await my custom task to complete before closing the snackbar, this could give the impression that the click did not register or the app has hung.

---

@henon Thoughts on this behavior? Makes sense to me.

---

Exceptions in the OnClick method would be swallowed though. So the best thing to do would be this:

`_ = InvokeAndLogException(State.Options.OnClick);` where `InvokeAndLogException` is:

```c#
[Inject]
private ILoggerFactory LoggerFactory { get; set; } = null!;
private ILogger? _logger;
protected ILogger Logger => _logger ??= LoggerFactory.CreateLogger(GetType());


private async Task InvokeAndLogException(Func<Snackbar, Task> func) {
  try {
    await func(this)
  }
  catch(Exception e) {
    Logger.LogErrpr("An exception happened in Snackbar.OnClick: {e.Message}\n");
  }
}
```

---

or maybe @ScarletKuro has a better idea

---

`Logger.LogError` has an overload that accepts `Exception?` that might provide stack as well.

---

Why it's simply not
```C#
await State.Options.OnClick.Invoke(this);
```
?
It's an `Func<Snackbar, Task>` and if re are renaming, it probably makes sense to name it `OnClickAsync`
Otherwise I don't get the `Clicked` change from `void` to `Task`
Well, I mean I understand you did it for the `ActionClickedAsync` / `CloseIconClickedAsync`, but it makes sense to use await now when it's a Task.